source/common/network/utility.cc

```diff
@@ -547,7 +550,59 @@ Api::IoCallUint64Result Utility::readFromSocket(IoHandle& handle,
                                                 UdpPacketProcessor& udp_packet_processor,
                                                 MonotonicTime receive_time,
                                                 uint32_t* packets_dropped) {
+
+  if (handle.supportsUdpGro()) {
+    Buffer::InstancePtr buffer = std::make_unique<Buffer::OwnedImpl>();
+    Buffer::RawSlice slice;
+
+    // TODO(yugant): Avoid using 64k by getting memory from UdpPacketProcessor
+    const uint64_t maxPacketSizeWithGro = 65535;
+    const uint64_t num_slices = buffer->reserve(maxPacketSizeWithGro, &slice, 1);
+    ASSERT(num_slices == 1u);
+
+    IoHandle::RecvMsgOutput output(1, packets_dropped);
+
+    Api::IoCallUint64Result result =
+        handle.recvmsg(&slice, num_slices, local_address.ip()->port(), output);
+
+    if (!result.ok()) {
+      return result;
+    }
+
```

Comment:
Upto this point, the code is very similar to recvmsg() without GRO below. Can you extract the shared code into a common helper function?

Reply:
Introduced a new function receiveMessage(...) and moved the common code into that!